server.js

```diff
@@ -10,6 +12,138 @@ app.use(express.static(__dirname + '/client'));
 /*app.get('/', function (req, res) {
   res.send('Hello World!');
 });*/
+
+// var config = require('./oauth.js');
+
+var passport = require('passport');
+var cookieParser = require('cookie-parser');
+app.use(cookieParser());
+// app.use(express.cookieParser());
+var bodyParser = require('body-parser');
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+var session = require('express-session');
+app.use(session({secret:'mysecret'}))
+app.use(passport.initialize())
+app.use(passport.session()); 
+
+var GoogleStrategy = require('passport-google-oauth2').Strategy;
+
+
+// create a user model
+
+var Schema = mongoose.Schema;
+
+var userSchema = new Schema({
+    oauthID: String,
+    name: String,
+    created: Date,
+    firstName: String,
+    lastName: String
+});
+
+var User = mongoose.model('User', userSchema);
+
+// serialize and deserialize
+    // used to serialize the user for the session
+    passport.serializeUser(function(user, done) {
+        done(null, user.id);
+    });
+
+    // used to deserialize the user
+    passport.deserializeUser(function(id, done) {
+        User.findById(id, function(err, user) {
+            done(err, user);
+        });
+    });
+
+// config
+
+var GoogleStrategy = require('passport-google-oauth2').Strategy;
+
+passport.use(new GoogleStrategy({
+  clientID: process.env.GOOGLE_CLIENT_ID,
+  clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+  callbackURL: "http://localhost:3000/auth/google/callback"
```

Comment:
Change this to `process.env.GOOGLE_AUTH_CALLBACK_URL` 
and fallback to `"http://localhost:'' + port + "/auth/google/callback"`